### PR TITLE
Generate the gardenlet landscaper blueprint

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
Replaces the manually created blueprint with an auto-generated one that is based on the golang types.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Based on #5233 and will be converted to a regular PR once this PR is merged.

**Release note**:

```other operator
The gardenlet blueprint is now auto-generated and is executed as part of running `make generate`
```

### CODE REVIEW

File: landscaper/common/generate/openapi/generate.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+	inputPackages := sets.NewString()
+	for _, v := range foo {
+		for _, valueFile := range v.Files {
+			for _, i := range valueFile.Imports {
+				escapedPackage := strings.ReplaceAll(i.Path.Value, "\"", "")
+				isK8sPackage := strings.HasPrefix(escapedPackage, "k8s.io")
+				isGithubPackage := strings.HasPrefix(escapedPackage, "github.com")
+
+				if !isGithubPackage && !isK8sPackage {
```
Comment (achimweigel) [inline landscaper/common/generate/openapi/generate.go]: Why are you skipping other packages?

**Review by achimweigel — COMMENTED** [review]
Looks really good to me. I will open an issue to add similar coding to the landscaper-cli such that also other projects could use it.

File: landscaper/common/generate/openapi/generate.go (hunk 1 above)
Comment (danielfoehrKn) [reply landscaper/common/generate/openapi/generate.go]: Mainly to speed up the generation process. This openapi annotation on the golang types are kube-openapi specific, hence Kubernetes specific.
If we find packages in the future that contain K8s types that are not either under `k8s.io` or `github.com` we can easily add a flag/argument to the program later. 
Does that sound ok?

Comment (achimweigel) [reply landscaper/common/generate/openapi/generate.go]: perfect :-)

**Comment by danielfoehrKn** [comment]
@rfranzke this is a follow-up PR to do the same kind of landscaper resource generation also for the controlplane.
The landscaper colleagues have already reviewed the PR it seems

**Review by rfranzke — APPROVED** [review]

**Pull request merged**